solidity/test/integration/Slashing.test.ts

```diff
@@ -0,0 +1,531 @@
+          txOutputIndex: 0,
+          txOutputValue: 18_500, // Value obtained from SingleP2SHDeposit.sweepTx.outputVector
+        }
+
+        // Request redemption
+        const redeemer = await helpers.account.impersonateAccount(
+          deposit.reveal.depositor,
+          { from: deployer }
+        )
+        const redemptionAmount = 3_000
+        redeemerOutputScript =
+          "0x17a91486884e6be1525dab5ae0b451bd2c72cee67dcf4187"
+
+        // Request redemption via TBTC Vault.
+        await tbtc
+          .connect(redeemer)
+          .approveAndCall(
+            tbtcVault.address,
+            redemptionAmount,
+            ethers.utils.defaultAbiCoder.encode(
+              ["address", "bytes20", "bytes32", "uint32", "uint64", "bytes"],
+              [
+                redeemer.address,
+                walletPubKeyHash160,
+                newMainUtxo.txHash,
+                newMainUtxo.txOutputIndex,
+                newMainUtxo.txOutputValue,
+                redeemerOutputScript,
+              ]
+            )
+          )
+
+        // Confirm the wallet is still in Live state.
+        expect(
+          (await await bridge.wallets(walletPubKeyHash160)).state
+        ).to.be.equal(walletState.Live)
+      })
+
+      describe("when a redemption timeout is reported", async () => {
+        let notifyRedemptionTimeoutTx: ContractTransaction
+
+        before(async () => {
+          const { redemptionTimeout } = await bridge.redemptionParameters()
+
+          await helpers.time.increaseTime(redemptionTimeout)
+
+          notifyRedemptionTimeoutTx = await bridge
+            .connect(thirdParty)
+            .notifyRedemptionTimeout(
+              walletPubKeyHash160,
+              walletMembers.getIds(),
+              redeemerOutputScript
+            )
+        })
+
+        it("should slash wallet members", async () => {
+          const { redemptionTimeoutSlashingAmount: amountToSlash } =
+            await bridge.redemptionParameters()
+
+          expect(await staking.getSlashingQueueLength()).to.equal(
+            walletMembers.length
+          )
+
+          for (let i = 0; i < walletMembers.length; i++) {
+            const slashing = await staking.slashingQueue(i)
+
+            expect(slashing.amount).to.equal(
+              amountToSlash,
+              `unexpected slashing amount for ${i}`
+            )
+
+            expect(slashing.stakingProvider).to.equal(
+              walletMembers[i].stakingProvider,
+              `unexpected staking provider for ${i}`
+            )
+          }
+        })
+
+        it("should not close the wallet in the wallet registry", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(await walletRegistry.isWalletRegistered(ecdsaWalletID)).to.be
+            .true
+        })
+
+        it("should transition the wallet in the bridge to moving funds state", async () => {
```

Comment:
Let's use the original state name as we usually do in unit tests:
```suggestion
        it("should transition the wallet in the bridge to the MovingFunds state", async () => {
```

Also, worth dropping a comment where we will explain that the wallet was switched to `MovingFunds` state because its BTC balance was non-zero.

Reply:
d21abed2d94e7f48ec126fefca9149ab2accbe3f